src/Parser/Mage/Frost/Modules/Cooldowns/ColdSnap.js

```diff
@@ -0,0 +1,25 @@
+import SPELLS from 'common/SPELLS';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import Analyzer from 'Parser/Core/Analyzer';
+import SpellUsable from 'Parser/Core/Modules/SpellUsable';
+
+class ColdSnap extends Analyzer {
+
+	static dependencies = {
+		combatants: Combatants,
+		spellUsable: SpellUsable,
+	}
+
+  	on_byPlayer_cast(event) {
+		const spellId = event.ability.guid;
+		if(spellId !== SPELLS.COLD_SNAP.id) {
+			return;
+		}
+		if (this.spellUsable.isOnCooldown(SPELLS.ICE_BARRIER.id)) { this.spellUsable.endCooldown(SPELLS.ICE_BARRIER.id); }
```

Comment:
I'd suggest making an array of the spell IDs and iterating over that to avoid all the if statements.